src/core/crypto/crypto_platform_psa.cpp

```diff
@@ -0,0 +1,793 @@
+    {
+        return error;
+    }
+
+    return psaToOtError(status);
+}
+
+OT_TOOL_WEAK otError otPlatCryptoExportKey(otCryptoKeyRef aKeyRef, uint8_t *aBuffer, size_t aBufferLen, size_t *aKeyLen)
+{
+    Error error = kErrorNone;
+
+    VerifyOrExit(aBuffer != nullptr && aKeyLen != nullptr, error = kErrorInvalidArgs);
+
+    error = psaToOtError(psa_export_key(aKeyRef, aBuffer, aBufferLen, aKeyLen));
+
+exit:
+    return error;
+}
+
+OT_TOOL_WEAK otError otPlatCryptoDestroyKey(otCryptoKeyRef aKeyRef) { return psaToOtError(psa_destroy_key(aKeyRef)); }
+
+OT_TOOL_WEAK bool otPlatCryptoHasKey(otCryptoKeyRef aKeyRef)
+{
+    psa_key_attributes_t attributes = PSA_KEY_ATTRIBUTES_INIT;
+    psa_status_t         status;
+
+    status = psa_get_key_attributes(aKeyRef, &attributes);
+    psa_reset_key_attributes(&attributes);
+
+    return status == PSA_SUCCESS;
+}
+
+OT_TOOL_WEAK otError otPlatCryptoAesInit(otCryptoContext *aContext)
+{
+    Error         error = kErrorNone;
+    psa_key_id_t *keyRef;
+
+    VerifyOrExit(aContext != nullptr, error = kErrorInvalidArgs);
+
+    keyRef  = static_cast<psa_key_id_t *>(aContext->mContext);
+    *keyRef = PSA_KEY_ID_NULL;
+
+exit:
+    return error;
+}
+
+OT_TOOL_WEAK otError otPlatCryptoAesSetKey(otCryptoContext *aContext, const otCryptoKey *aKey)
+{
+    Error         error = kErrorNone;
+    psa_key_id_t *keyRef;
+
+    VerifyOrExit(checkContext(aContext, sizeof(psa_key_id_t)), error = kErrorInvalidArgs);
+    VerifyOrExit(aKey != nullptr, error = kErrorInvalidArgs);
+
+    keyRef  = static_cast<psa_key_id_t *>(aContext->mContext);
+    *keyRef = aKey->mKeyRef;
+
+exit:
+    return error;
+}
+
+OT_TOOL_WEAK otError otPlatCryptoAesEncrypt(otCryptoContext *aContext, const uint8_t *aInput, uint8_t *aOutput)
+{
+    Error         error     = kErrorNone;
+    const size_t  blockSize = PSA_BLOCK_CIPHER_BLOCK_LENGTH(PSA_KEY_TYPE_AES);
+    psa_status_t  status    = PSA_SUCCESS;
+    psa_key_id_t *keyRef;
+    size_t        cipherLen;
+
+    VerifyOrExit(checkContext(aContext, sizeof(psa_key_id_t)), error = kErrorInvalidArgs);
+    VerifyOrExit(aInput != nullptr && aOutput != nullptr, error = kErrorInvalidArgs);
+
+    keyRef = static_cast<psa_key_id_t *>(aContext->mContext);
+    status = psa_cipher_encrypt(*keyRef, PSA_ALG_ECB_NO_PADDING, aInput, blockSize, aOutput, blockSize, &cipherLen);
+
+    error = psaToOtError(status);
+
+exit:
+    return error;
+}
+
+OT_TOOL_WEAK otError otPlatCryptoAesFree(otCryptoContext *aContext)
+{
+    OT_UNUSED_VARIABLE(aContext);
+
+    return kErrorNone;
+}
+
+#if !OPENTHREAD_RADIO
+
+OT_TOOL_WEAK otError otPlatCryptoHmacSha256Init(otCryptoContext *aContext)
+{
+    Error                error = kErrorNone;
+    psa_mac_operation_t *operation;
+
+    VerifyOrExit(checkContext(aContext, sizeof(psa_mac_operation_t)), error = kErrorInvalidArgs);
+
+    operation  = static_cast<psa_mac_operation_t *>(aContext->mContext);
+    *operation = psa_mac_operation_init();
```

Comment:
Suggest using `memset(operation, 0, sizeof(*operation)` instead to reduce the stack usage.